interop/README.md

```diff
@@ -0,0 +1,49 @@
+# Interop Test Runner
+
+The Interop Test Runner aims to automatically generate an interop matrix by running multiple **test cases** using different QUIC implementations.
+
+## Running the Interop Runner
+
+The Interop Runner is written in Python 3. You'll need to install a few Python modules to run it:
+
+```bash
+pip3 install termcolor prettytable pyshark
+```
+
+Furthermore, you need to install Wireshark (version 3.0.5 or newer).
+
+Run the interop tests:
+```bash
+python3 run.py
+```
+
+## Building a QUIC endpoint
+
+To include your QUIC implementation in the Interop Runner, create a Docker image, publish it on [https://hub.docker.com](Docker Hub) and add it to [implementations.py](implementations.py). Once your implemetation is ready to interop, please send us a PR with this addition.
+
+Typically, a test case will require a server to serve files from a directory, and a client to download files. Different test cases will specify the behavior to be tested. For example, the Retry test case expects the server to use a Retry before accepting the connection from the client.
+The test case is passed into your Docker container using the `TESTCASE` environment variable. If your implementation doesn't support a test case, it MUST exit with status code 127. This will allow us to add new test cases in the future, and correctly report test failures und successes, even if some implementations have not yet implented support for this new test case.
+
+The Interop Runner mounts the directory `/www` into your server Docker container. This directory will contain one or more randomly generated files. Your server implementation is expected to run on port 443 and serve files from this directory.
+Equivalently, the Interop Runner mounts `/downloads` into your client Docker container. The directory is initially empty, and your client implementation is expected to store downloaded files into this directory. The URLs of the files to download are passed to the client using the environment variable `REQUESTS`, which contains one or more URLs, separated by a space.
+
+After the transfer is completed, the client container is expected to exit with exit status 0. If an error occurred during the transfer, the client is expected to exit with exit status 1.
```

Comment:
You should add that if possible, implementations should call it an error if the particular test case was not correctly executed by the peer. For example, for a resumption test case, if the client did not provide a token on the second connection attempt.